File: client-app/src/app/models/activity.ts
import { Profile } from "./profile";

export interface iActivity {
    id: string;
    title: string;
    date: Date | null;
    description: string;
    category: string;
    city: string;
    venue: string; 
    hostUsername:string;
    isCancelled:boolean;
    isGoing:boolean;
    isHost:boolean;
    host?:Profile;
    attendees:Profile[]
}


export class iActivity implements iActivity{
    constructor(init?: ActivityFormValues){
        Object.assign(this.attendees, init);
    }
}

export class ActivityFormValues{
    id?: string = undefined;
    title: string = '';
    category: string = '';
    description: string = '';
    date: Date | null = null;
    city: string = '';
    venue: string = '';

    constructor(activity?: ActivityFormValues){
         if(activity){
            this.id = activity.id;
            this.title = activity.title;
            this.category = activity.category;
            this.description = activity.description;
            this.date = activity.date;
            this.city = activity.city;
            this.venue = activity.venue;
         }
    } 
}

File: Domain/Activity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Activity
    {
        public Guid Id { get; set; }


        public string Title { get; set; }

        public  DateTime date { get; set; }

        public string Description { get; set; }

        public string Category { get; set; }

        public string City { get; set; }

        public string Venue { get; set; }

        public bool isCancelled {get; set;}

        public ICollection<ActivityAttendee> Attendees{get; set;}=new List <ActivityAttendee>();

        public ICollection<Comment> Comments{get; set;} = new List<Comment>();
    }
}